modules/appeals_api/app/models/concerns/appeals_api/validation.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module AppealsApi
+  module Validation
+    extend ActiveSupport::Concern
+
+    included do
+      private
+
+      # validation (header & body)
+      # Schemas take care of most of the requirements, but we need to check that both header & body data is provided
+      def required_claimant_data_is_present
+        has_claimant_headers = claimant.first_name.present?
+        # form data that includes a claimant is also sufficient to know it's passed the schema
+        has_claimant_data = data_attributes&.fetch('claimant', nil).present?
+
+        return if !has_claimant_headers && !has_claimant_data # No claimant headers or data? not a problem!
+        return if has_claimant_headers && has_claimant_data # Has both claimant headers and data? A-ok!
+
+        unless has_claimant_headers
+          add_error('Claimant data was provided but missing claimant headers',
+                    missing_fields: missing_fields(auth_headers, form_data),
+                    title: 'Missing required fields',
+                    pointer: '/', code: 145)
+        end
+        unless has_claimant_data
+          add_error('Claimant headers were provided but missing claimant data',
+                    missing_fields: missing_fields(auth_headers, form_data),
```

Comment:
I think in this case we should combine AddError and Validation concerns, especially since Validation relies so heavily on things AddError provides. Let's combine the two files into one Validation concern will work, rather than trying to include AddError into Validation or any concern onion-ing trickery :)